dotnet-desktop-guide/framework/winforms/controls/walkthrough-debugging-custom-windows-forms-controls-at-design-time.md

```diff
@@ -23,6 +23,9 @@ When you create a custom control, you will often find it necessary to debug its
 
 You can debug your custom controls using Visual Studio, just as you would debug any other .NET Framework classes. The difference is that you will debug a separate instance of Visual Studio that is running your custom control's code.
 
+> [!IMPORTANT]
+> This article is primarily intended for the classic In-Process Designer for Windows Forms with .NET Framework. The procedures may not be directly applicable or fully compatible with projects that target .NET. Further, Visual Studio 2022 runs on .NET Framework 4.7.2 in a 64-bit environment, which can cause compatibility issues with Out-Of-Process Designers if you target a 32-bit architecture with your .NET applications. For more information about the Out-Of-Process Designer, see [The designer changes since .NET Framework](../../../net/winforms/controls-design/designer-differences-framework.md).
```

Comment:
```suggestion
> This article is primarily intended for the classic In-Process Designer for Windows Forms with .NET Framework. The procedures may not be directly applicable or fully compatible with projects that target .NET (Core 3.1 or 5.0+). Further, Visual Studio 2022 runs on .NET Framework 4.7.2 in a 64-bit environment, which can cause compatibility issues with WinForms projects which rely on ActiveX controls or COM or other components, that can only run in a 32-Bit process. The Out-Of-Process _Framework_ Designer addresses most of those issues, but design-time functionality may - as is true for the .NET Out-Of-Process Designer - not be compatible with the classic in-proc Designer. For more information about the Out-Of-Process Designer, see [The designer changes since .NET Framework](../../../net/winforms/controls-design/designer-differences-framework.md).

**Note:** The Out-Of-Process Framework Designer is currently in Preview, and can be activated in Preview versions of Visual Studio via _Tools/Options/Environment/Preview Features_.
```

Reply:
I'm a little confused. You used **out-of-process framework designer** and **out-of-process designer** which is correct? Is there two out-of-process designers??

Reply:
```suggestion
> This article is primarily intended for the classic In-Process Designer for Windows Forms with .NET Framework. The procedures may not be directly applicable or fully compatible with projects that target .NET (not .NET Framework).
>
> For .NET Framework projects, only projects that compile to 64-bit are fully supported. Projects that use ActiveX, COM, or other 32-bit components, may have compatibility issues. The Out-Of_Process Framework Designer, currently in preview, addresses many of those compatibility issues. However, design-time functionality may not work as well as the 64-bit In-Process Designer. To enable the Out-Of-Process Framework Designer, navigate to the **Tools** > **Options** > **Environment** > **Preview Features**.
>
> For more information about the Out-Of-Process Designer, see [The designer changes since .NET Framework](../../../net/winforms/controls-design/designer-differences-framework.md).
```

Reply:
Perfect!

Reply:
Ah, wait...no, I need to rephrase a bit...

Reply:
```suggestion
> This article is primarily intended for the classic In-Process Designer for Windows Forms with .NET Framework. The procedures may not be directly applicable or fully compatible with projects that target .NET (not .NET Framework).
>
> For .NET Framework projects, only projects that can compile with the platform target settings _Any_ or _x64_ are fully supported in the Designer. Projects that use ActiveX, COM, or other 32-bit components, may have compatibility issues. The Out-Of_Process Framework Designer, currently in preview, addresses many of those compatibility issues. However, design-time functionality may not work as well as the 64-bit In-Process Designer. To enable the Out-Of-Process Framework Designer, navigate to the **Tools** > **Options** > **Environment** > **Preview Features**.
>
> For more information about the Out-Of-Process Designer, see [The designer changes since .NET Framework](../../../net/winforms/controls-design/designer-differences-framework.md).
```